### DISCUSSION TITLE
xUnit1041: fix for scenarios involving inheritance

### DISCUSSION THREAD

**Discussion by felpel** [xunit/xunit#2847] [General]
Hi 👋 

The [`xUnit1041` error](https://xunit.net/xunit.analyzers/rules/xUnit1041) is now being reported following the upgrade of `xunit` to `2.6.3`.

Based on [this comment](https://github.com/xunit/xunit/issues/2311#issuecomment-1848384304) and [this one](https://github.com/xunit/xunit/issues/2311#issuecomment-1848674176), is the `1.8.0` version - or any other version with aforementioned fix - of `xunit.analyzers` in the works/soon to be published? 🤞 

If possible, it would be neat to avoid the following mitigations: 
* disable the analyzer rule
* target a pre-release version of `xunit.analyzers`
* decorate concrete test classes with `CollectionAttribute` rather than the base class.  

Thank you for your time!

**Comment by bradwilson** [comment]
Yes, as you've linked to, a prerelease build exists with the fix: `1.8.0-pre.2`. https://xunit.net/docs/using-ci-builds

**Reply by felpel** [reply]
As I've said above:
> If possible, it would be neat to avoid the following mitigations: [...] target a pre-release version of xunit.analyzers

I will rephrase my original question: can we expect a new patch release soon to include this fix?

I have no issue patching my original code, but I would prefer not to modify my dependencies, the `NuGet` config or the EditorConfig config if the code is to be released in less than a week or so 😅

**Reply by bradwilson** [reply]
> I would prefer not to modify my dependencies, the NuGet config or the EditorConfig config if the code is to be released in less than a week or so

I can't tell you when it'll be released yet. For now you're going to have to roll back to 2.6.2 because I can't provide any solution with your restricted parameters.

**Comment by fdonnet** [comment]
Thx for the report here, it allowed me to not loose time when I upgraded xUnit.
